### PR TITLE
Wi 54 edit user profile edward

### PR DESCRIPTION
**Pull request by Ed-C03**

### CODE REVIEW

File: pages/edituserprofile.tsx [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+});
+
+// Define a component that renders the form
+export default function EditUserProfile() {
+  const router = useRouter();
+  const methods = useForm<IFormInput>({
+    resolver: yupResolver(formSchema),
+  });
+
+  const [LastName, setLastName] = useState("");
```
Comment (Chanran) [inline pages/edituserprofile.tsx]: lastName

File: pages/edituserprofile.tsx [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  const [FirstName, setFirstName] = useState("");
```
Comment (Chanran) [inline pages/edituserprofile.tsx]: firstName

File: pages/edituserprofile.tsx [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+  const [avatar, setAvatar] = useState("");
+
+  useEffect(() => {
+    axiosRequest(`/api/editUser/me`, "GET").then((res) => {
```
Comment (Chanran) [inline pages/edituserprofile.tsx]: move request api code to api folder

File: pages/edituserprofile.tsx [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+      setLastName(res.data.lastName);
+      setFirstName(res.data.firstName);
+      setAvatar(res.data.avatarPath);
+    });
+  }, [avatar]);
+
+  // Define a submit handler for the form
+  const onSubmit: SubmitHandler<IFormInput> = async (data: IFormInput) => {
+    try {
+      const response = await axiosRequest(
```
Comment (Chanran) [inline pages/edituserprofile.tsx]: Move request api code to api folder

File: pages/edituserprofile.tsx [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+        "/api/editUser/updateName",
+        "PATCH",
+        data
+      );
+
+      if (response.status === 200) {
+        router.push("/edituserprofile");
+      }
+    } catch (error) {
+      alert("Error occurred, unknown origin.");
```
Comment (Chanran) [inline pages/edituserprofile.tsx]: use mui Alert or Snackbar Component

**Review by Adamyyyy — APPROVED** [review]
LGTM

**Pull request merged**